pkgs/desktops/lomiri/qml/lomiri-ui-toolkit/default.nix

```diff
@@ -0,0 +1,337 @@
+    for badQmlTest in ${lib.strings.concatStringsSep " " [
+      # Requires a working Qt OpenGL context in ShapeMaterial
+      "tst_actionbar.13.qml"
+      "tst_actionlist.13.qml"
+      "tst_adaptivepagelayout.13.qml"
+      "tst_adaptivepagelayout_configuration.13.qml"
+      "tst_combobutton.11.qml"
+      "tst_combobutton.13.qml"
+      "tst_contextual_actions.13.qml"
+      "tst_focus.13.qml"
+      "tst_header.13.qml"
+      "tst_hide_chrome.11.qml"
+      "tst_listitem.12.qml"
+      "tst_listitem.13.qml"
+      "tst_listitem_actions_breaks_selectmode.12.qml"
+      "tst_listitem_extras.13.qml"
+      "tst_listitem_focus.13.qml"
+      "tst_listitem_horizontal_navigation.13.qml"
+      "tst_listitem_selectmode.12.qml"
+      "tst_listitem_selectmode.13.qml"
+      "tst_listitems_itemselector.11.qml"
+      "tst_listitems_standard.11.qml"
+      "tst_listitems_standard.13.qml"
+      "tst_lomirilistview.11.qml"
+      "tst_lomiritestcase.qml"
+      "tst_multicolumnheader.13.qml"
+      "tst_optionselector.11.qml"
+      "tst_optionselector.13.qml"
+      "tst_page_with_header.13.qml"
+      "tst_pagehead_back_action.13.qml"
+      "tst_pagehead_contents_width.13.qml"
+      "tst_pagehead_sections.13.qml"
+      "tst_pagehead_visible.13.qml"
+      "tst_pageheader.13.qml"
+      "tst_pagestack.13.qml"
+      "tst_pagestack.DEPRECATED_APPHEADER_TABS.13.qml"
+      "tst_picker.11.qml"
+      "tst_picker.13.qml"
+      "tst_popover.12.qml"
+      "tst_popover.13.qml"
+      "tst_popups_dialog.13.qml"
+      "tst_popups_pagestack.13.qml"
+      "tst_pulltorefresh_pagestack_topmargin.13.qml"
+      "tst_slider.11.qml"
+      "tst_slider.13.qml"
+      "tst_switch_bug1510919.13.qml"
+      "tst_tabs.11.qml"
+      "tst_tabs.13.qml"
+      "tst_tabs.DEPRECATED_TOOLBAR.11.qml"
+      "tst_textarea.11.qml"
+      "tst_textarea_in_flickable.11.qml"
+      "tst_textfield.11.qml"
+      "tst_textinput_common.12.qml"
+      "tst_textinput_common.13.qml"
+      "tst_toggles.13.qml"
+
+      # SignalSpy on QML shader compilers are failing for unknown reasons
+      "tst_icon.11.qml"
+      "tst_icon.13.qml"
+    ]}; do
+      find tests/unit -name $badQmlTest -delete
+    done
+
+    # These tests are registered explicitly, just removing them from the list is enough
+    for badUnitTest in ${lib.strings.concatStringsSep " " [
+      # Requires a working Qt OpenGL context in ShapeMaterial
+      "inversemousearea"
+      "layouts"
+      "recreateview"
+      "subtheming"
+      "swipearea"
+    ]}; do
+      sed -i -e "/$badUnitTest/d" tests/unit/unit.pro
+    done
+
+    # These tests don't fail explicitly, but issue warnings that the test suite doesn't expect, which it turns into errors
+    for unexpectedWarningTest in ${lib.strings.concatStringsSep "" [
+      # qml/Lomiri/Components/1.2/Toolbar.qml:96: TypeError: Cannot read property 'locked' of null
+      # I don't know QML enough to fix this, upstream supposedly knows about this already
```

Comment:
> upstream supposedly knows about this already

can you add a link?

Reply:
The note was from [a Debian patch](https://salsa.debian.org/ubports-team/lomiri-ui-toolkit/-/blob/362d58baddb1d9c818d1846eaaa59b6390fee550/debian/patches/2002_exclude-dodgy-tests-for-now.patch#L3), I think this refers to https://gitlab.com/ubports/development/core/lomiri-ui-toolkit/-/issues/9. Will add the link.